pkg/operations/kubernetesupgrade/upgrademasternode.go

```diff
@@ -72,5 +75,51 @@ func (kmn *UpgradeMasterNode) CreateNode(poolName string, masterNo int) error {
 
 // Validate will verify the that master/agent node has been upgraded as expected.
 func (kmn *UpgradeMasterNode) Validate(vmName *string) error {
-	return nil
+	if vmName == nil || *vmName == "" {
+		kmn.logger.Warningf("VM name was empty. Skipping node condition check")
+		return nil
+	}
+
+	var masterURL string
+	if kmn.UpgradeContainerService.Properties.HostedMasterProfile != nil {
+		masterURL = kmn.UpgradeContainerService.Properties.HostedMasterProfile.FQDN
```

Comment:
If this is hostedmasterprofile then there is no node to validate or this code path won't even be it. Can you validate?